pkg/proxy/userspace/proxier.go

```diff
@@ -687,12 +696,16 @@ func iptablesCommonPortalArgs(destIP net.IP, destPort int, protocol api.Protocol
 		args = append(args, "-d", fmt.Sprintf("%s/32", destIP.String()))
 	}
 
+	if excludeContainerSourceTraffic {
+		args = append(args, "-m", "physdev", "--physdev-is-in")
```

Comment:
yes, will try it now

On Sat, Aug 15, 2015 at 9:54 PM, Tim Hockin <notifications@github.com>
wrote:

> In pkg/proxy/userspace/proxier.go
> <https://github.com/kubernetes/kubernetes/pull/12561#discussion_r37142001>
> :
>
> > @@ -687,12 +696,16 @@ func iptablesCommonPortalArgs(destIP net.IP, destPort int, protocol api.Protocol
> >  		args = append(args, "-d", fmt.Sprintf("%s/32", destIP.String()))
> >  	}
> >
> > +	if excludeContainerSourceTraffic {
> > +		args = append(args, "-m", "physdev", "--physdev-is-in")
>
> Ahh, I see you have named a variable backwards. This flag is called
> "exclude" but you pass false to it when you want to exclude. Try it the
> other way.
>
> You only care about whether it is a bridged packet IFF the caller said to
> exclude.
>
> clusterIP: don't care about bridging
> load-balancer: don't care about bridging
> external IP: don't allow bridging
>
> I think that handles what you want here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/12561/files#r37142001>.
>